lib/p5.js

```diff
@@ -4661,6 +4672,34 @@ amdclean['helpers'] = function (require, core) {
       }
     }
   };
+  p5._friendlyFileLoadError = function (errorType, filePath) {
+    switch (errorType) {
+    case 0:
+      var reportMsg0 = 'It looks like there was a problem' + ' loading your image.' + ' Try checking if the file path%c [' + filePath + '] %cis correct,' + ' hosting the image online, or running a local server.';
+      report(reportMsg0, 'loadImage', FILE_LOAD);
+      break;
+    case 1:
+      var reportMsg1 = 'It looks like there was a problem' + ' loading your XML file.' + ' Try checking if the file path%c [' + filePath + '] %cis correct,' + ' or running a local server.';
+      report(reportMsg1, 'loadXML', FILE_LOAD);
+      break;
+    case 2:
+      var reportMsg2 = 'It looks like there was a problem' + ' loading your table file.' + ' Try checking if the file path%c [' + filePath + '] %cis correct,' + ' or running a local server.';
+      report(reportMsg2, 'loadTable', FILE_LOAD);
+      break;
+    case 3:
+      var reportMsg3 = 'It looks like there was a problem' + ' loading your text file.' + ' Try checking if the file path%c [' + filePath + '] %cis correct,' + ' or running a local server.';
+      report(reportMsg3, 'loadStrings', FILE_LOAD);
+      break;
+    default:
```

Comment:
The use of a switch here makes this feel slightly duplicative. I typed up an alternative, using a dictionary object to reduce the duplication of the message string creation:
```js
var errorCases = {
  '0': {
    fileType: 'image',
    method: 'loadImage',
    message: ' hosting the image online,'
  },
  '1': {
    fileType: 'XML file',
    method: 'loadXML'
  },
  '2': {
    fileType: 'table file',
    method: 'loadTable'
  },
  '3': {
    fileType: 'text file',
    method: 'loadStrings'
  }
};
p5._friendlyFileLoadError = function (errorType, filePath) {
  var errorInfo = errorCases[ errorType ];
  var message = 'It looks like there was a problem' +
    ' loading your ' + errorInfo.fileType + '.' +
    ' Try checking if the file path%c [' + filePath + '] %cis correct,' +
    (errorInfo.message || '') + ' or running a local server.';
    report(message, errorInfo.method, FILE_LOAD);
};
```
however, this is un-tested and it's arguably less intuitive, so I present it only as food for thought, for now. :)